src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheckTest.java

```diff
@@ -0,0 +1,165 @@
+        verify(checkConfig, getPath("InputMatchXpath.java"), expected);
+    }
+
+    @Test
+    public void testCheckWithDoubleBraceInitialization()
+            throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(MatchXpathCheck.class);
+
+        checkConfig.addAttribute("query", "//INSTANCE_INIT[not(../*[not(\n"
+                + "                    self::LCURLY or\n"
+                + "                    self::INSTANCE_INIT or\n"
+                + "                    self::RCURLY or\n"
+                + "                    self::SINGLE_LINE_COMMENT or\n"
+                + "                    self::BLOCK_COMMENT_BEGIN\n"
+                + "                )])]");
+        checkConfig.addMessage("matchxpath.match", "Do not use double-brace initialization");
+        final String[] expected = {
+            "11:35: Do not use double-brace initialization",
+        };
+        verify(checkConfig, getPath("InputMatchXpathDoubleBrace.java"), expected);
+    }
+
+    @Test
+    public void testImitateIllegalThrowsCheck()
+            throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(MatchXpathCheck.class);
+        checkConfig.addAttribute("query", "//LITERAL_THROWS[./IDENT[@text='Throwable' or "
+                + "@text='RuntimeException' or ends-with(@text, 'Error')]]");
+        checkConfig.addMessage("matchxpath.match", "Illegal throws statement");
+        final String[] expected = {
+            "8:25: Illegal throws statement",
+            "10:25: Illegal throws statement",
+            "11:25: Illegal throws statement",
+        };
+        verify(checkConfig, getPath("InputMatchXpathIllegalThrows.java"), expected);
+    }
+
+    @Test
+    public void testImitateExecutableStatementCountCheck()
+            throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(MatchXpathCheck.class);
+        checkConfig.addAttribute("query", "//METHOD_DEF[count(./SLIST/*) > 2]");
+        checkConfig.addMessage("matchxpath.match", "Executable number of statements "
+                + "exceed threshold");
+        final String[] expected = {
+            "21:5: Executable number of statements exceed threshold",
+        };
+        verify(checkConfig, getPath("InputMatchXpathExecutableStatementCount.java"), expected);
+    }
+
+    @Test
+    public void testInvalidQuery() {
+        final MatchXpathCheck matchXpathCheck = new MatchXpathCheck();
+        matchXpathCheck.setQuery("!@#%^");
+
+        final DetailAstImpl detailAST = new DetailAstImpl();
+        detailAST.setType(TokenTypes.CLASS_DEF);
+        detailAST.setText("Class Def");
+        detailAST.setLineNo(0);
+        detailAST.setColumnNo(0);
+
+        try {
+            matchXpathCheck.beginTree(detailAST);
+            fail("Exception was expected");
+        }
+        catch (IllegalStateException ex) {
```

Comment:
`ex` -> `ignored`. This is common naming convention for such cases.

Another (preferred) option is to assert the exception message.

Reply:
done